index/compaction.go

```diff
@@ -316,3 +329,48 @@ func (l *inMemoryLevel) Compact(toCompact []parts.Part, options ...parts.Option)
 	postCompactionSize := int64(b.Len())
 	return []parts.Part{parts.NewParquetPart(0, buf, options...)}, preCompactionSize, postCompactionSize, nil
 }
+
+// ValidateIndexDir will validate each index file in the given directory. It will read the last part in each file and ensure that it can open it.
+func ValidateIndexDir(dir string) error {
+	return filepath.WalkDir(dir, func(path string, d os.DirEntry, err error) error {
+		if err != nil {
+			return fmt.Errorf("failed to walk snapshot index directory: %w", err)
+		}
+
+		if d.IsDir() { // Table and Level dirs expected
+			return nil
+		}
+
+		if filepath.Ext(path) != IndexFileExtension {
+			return nil // unknown file
+		}
+
+		idxfile, err := os.Open(path)
+		if err != nil {
+			return fmt.Errorf("failed to open index file %s: %w", path, err)
+		}
+		defer idxfile.Close()
+
+		// Attempt to read the final Parquet file to ensure it is valid.
+		info, err := idxfile.Stat()
+		if err != nil {
+			return fmt.Errorf("failed to get index file %s info: %w", path, err)
+		}
+
+		offset := info.Size() - 8
+		size := make([]byte, 8)
+		if n, err := idxfile.ReadAt(size, offset); n != 8 {
+			return fmt.Errorf("failed to read index file %s size: %w", path, err)
+		}
+
+		parquetSize := int64(binary.LittleEndian.Uint64(size))
+		offset -= parquetSize
+
+		_, err = parquet.OpenFile(io.NewSectionReader(idxfile, offset, parquetSize), parquetSize)
```

Comment:
I was wondering do we only care that the snapshot is a `parquet` file and not the actual file we actually snapshotted ?

I can be wrong, but when validating backup, the need to know you are working with the actual data you backup is important.

I'm talking about having a checksum of the written data something like `crc32` checksum.

You can completely ignore this comment, it is random question rather than a review.

Reply:
We're less concerned that this is the file we expect and more concerned that it's a valid file. That said I'm going to close this for now because I want to refactor snapshots as they are right now as it's a little convoluted between the table indices and the snapshot file.